### PR TITLE
Design api test

### PR DESCRIPTION
**Pull request by AnnaKalinovskaya**

### CODE REVIEW

File: src/main/java/org/API test design [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+### Get auth token
+POST http://localhost:4111/api/v1/auth/login
+Content-Type: application/json
+{
+  "username": "hanna",
+  "password": "hannaPass1!"
+}
+
+### Positive test: User can deposit money to his bank account (valid boundary values)
```
Comment (nobugs-me) [inline src/main/java/org/API test design]: Отличный охват boundary-value анализом (0.01, 4999.99, 5000), но не хватает теста с максимально возможным значением (например, Long.MAX_VALUE), чтобы явно зафиксировать ограничение по типу данных. Добавь такой тест, чтобы поймать edge-кейсы.

File: src/main/java/org/API test design [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+### Negative test: User cannot deposit INVALID amount to his bank account (boundary values)
+POST http://localhost:4111/api/v1/admin/users
+Content-Type: application/json
+Authorization: Basic aGFubmE6aGFubmFQYXNzMSE=
+{
+  "id": 1,
+  "balance": 5000.01
+}
+
+### Negative test: User cannot deposit amount to his non-existing bank account (invalid id)
```
Comment (nobugs-me) [inline src/main/java/org/API test design]: Ты проверяешь невалидный id (id = 3), но почему именно 3? Если 3 может существовать в других случаях, тест станет хрупким. Лучше использовать рандомный большой id, который гарантированно не существует (например, 999999).

File: src/main/java/org/API test design [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+POST http://localhost:4111/api/v1/accounts/transfer
+Content-Type: application/json
+Authorization: Basic aGFubmE6aGFubmFQYXNzMSE=
+{
+  "senderAccountId": 1,
+  "receiverAccountId": 2,
+  "amount": 0.01
+}
+
+### Positive test: User can transfer money (boundary values) when sender bank account balance is at least 10 000
```
Comment (nobugs-me) [inline src/main/java/org/API test design]: Отлично, что проверяешь граничные условия. Но добавь тест, где sender и receiver — один и тот же аккаунт. Сейчас такого кейса нет, а это интересный баг-кандидат.

File: src/main/java/org/API test design [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+POST http://localhost:4111/api/v1/accounts/transfer
+Content-Type: application/json
+Authorization: Basic aGFubmE6aGFubmFQYXNzMSE=
+{
+  "senderAccountId": 1,
+  "receiverAccountId": 2,
+  "amount": 10 000
+}
+
+### Negative test: User cannot transfer INVALID amount (boundary values)
```
Comment (nobugs-me) [inline src/main/java/org/API test design]: Хорошо, что проверяешь несуществующие аккаунты. Но стоит добавить тест на несовпадающие валюты/тип счёта, если система поддерживает разные типы (см. бизнес-логику).

File: src/main/java/org/API test design [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+### Negative test: User cannot transfer money TO NON-existing account.
+POST http://localhost:4111/api/v1/accounts/transfer
+Content-Type: application/json
+Authorization: Basic aGFubmE6aGFubmFQYXNzMSE=
+{
+  "senderAccountId": 1,
+  "receiverAccountId": 3,
+  "amount": 1000
+}
```
Comment (nobugs-me) [inline src/main/java/org/API test design:132]: Для senior-уровня стоит добавить хотя бы один тест, который проверяет конкурентные переводы (например, два параллельных запроса на перевод с одного счёта), чтобы увидеть, есть ли проблема с балансом.

File: src/main/java/org/API test design [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+-----------------------------------UPDATE USER NAME-----------------------------------
+
+### Positive test: user can update his name with valid value (2 words, all letters)
```
Comment (nobugs-me) [inline src/main/java/org/API test design]: Молодец, что учитываешь разные некорректные форматы (1 слово, 3 слова, цифры, спецсимволы, пустое). Но нет теста на максимальную длину имени (например, 255 символов, если это лимит в БД). Добавь такой кейс, чтобы закрыть длину.

File: src/main/java/org/API test design [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+PUT http://localhost:4111/api/v1/customer/profile
+Content-Type: application/json
+Authorization: Basic aGFubmE6aGFubmFQYXNzMSE=
+{
+  "name": "Hanna Customer"
+}
+
+### Negative test: user cannot update his name with invalid name (1 word, all letters)
+PUT http://localhost:4111/api/v1/customer/profile
```
Comment (nobugs-me) [inline src/main/java/org/API test design]: Добавь тест, который отправляет PUT с тем же именем, что уже сохранено. Это поможет поймать ненужные обновления и проверить, что система адекватно реагирует.

**Review by nobugs-me — CHANGES_REQUESTED** [review]
Привет!

Спасибо за выполненное задание, крутой результат!

Что сделано круто:

Отличное покрытие boundary-value тестами

Есть и позитивные, и негативные сценарии, хорошо проверяешь бизнес-валидации

Учитываешь edge-кейсы типа несуществующих аккаунтов

Что нужно пофиксить:

Добавить тесты на экстремальные значения (максимальная сумма, максимальная длина имени)

Усилить тесты на race conditions или конкурентные запросы

Вынести дублирующие сценарии в параметризованные тесты для лучшей читаемости и поддержки

Добавить проверки на одинаковые значения (например, update без изменений)

Давай исправим комменты и будет аппрув.

File: src/main/java/org/API test design (hunk 1 above)
Comment (AnnaKalinovskaya) [reply src/main/java/org/API test design]: добавила значения как новые параметры для теста

File: src/main/java/org/API test design (hunk 2 above)
Comment (AnnaKalinovskaya) [reply src/main/java/org/API test design]: изменила

File: src/main/java/org/API test design (hunk 3 above)
Comment (AnnaKalinovskaya) [reply src/main/java/org/API test design]: добавила

File: src/main/java/org/API test design (hunk 4 above)
Comment (AnnaKalinovskaya) [reply src/main/java/org/API test design]: судя по требованиям и документации в свагере система с разными типами валют и аккаунтами не работает. Таких полей нет

File: src/main/java/org/API test design (hunk 5 above)
Comment (AnnaKalinovskaya) [reply src/main/java/org/API test design:132]: добавила для transfer и deposit

File: src/main/java/org/API test design (hunk 6 above)
Comment (AnnaKalinovskaya) [reply src/main/java/org/API test design]: добавила

File: src/main/java/org/API test design (hunk 7 above)
Comment (AnnaKalinovskaya) [reply src/main/java/org/API test design]: добавила

**Review by nobugs-me — APPROVED** [review]
Спасибо! Теперь зачтено
